error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/BadStaticImport.java

```diff
@@ -0,0 +1,194 @@
+   * StaticImport#STATIC_IMPORT_CANDIDATE_MEMBERS}.
+   *
+   * <p>Types listed here should be mutually exclusive with {@link
+   * StaticImport#STATIC_IMPORT_CANDIDATE_TYPES}
+   */
+  static final ImmutableSet<String> BAD_STATIC_IMPORT_CANDIDATE_TYPES =
+      ImmutableSet.of("com.google.common.base.Strings", "java.time.Clock", "java.time.ZoneOffset");
+
+  /**
+   * Type members that should never be statically imported.
+   *
+   * <p>Identifiers listed by {@link #BAD_STATIC_IMPORT_CANDIDATE_IDENTIFIERS} should be omitted
+   * from this collection.
+   *
+   * <p>This should be mutually exclusive with {@link StaticImport#STATIC_IMPORT_CANDIDATE_MEMBERS}
+   */
+  // XXX: Perhaps the set of exempted `java.util.Collections` methods is too strict. For now any
+  // method name that could be considered "too vague" or could conceivably mean something else in a
+  // specific context is left out.
+  static final ImmutableSetMultimap<String, String> BAD_STATIC_IMPORT_CANDIDATE_MEMBERS =
+      ImmutableSetMultimap.<String, String>builder()
+          .put("com.mongodb.client.model.Filters", "empty")
+          .putAll(
+              "java.util.Collections",
+              "addAll",
+              "copy",
+              "fill",
+              "list",
+              "max",
+              "min",
+              "nCopies",
+              "rotate",
+              "sort",
+              "swap")
+          .put("java.util.Locale", "ROOT")
+          .put("java.util.Optional", "empty")
+          .putAll("java.util.regex.Pattern", "compile", "matches", "quote")
+          .put("org.springframework.http.MediaType", "ALL")
+          .build();
+
+  /**
+   * Identifiers that should never be statically imported.
+   *
+   * <p>Identifiers listed by {@link StaticImport#STATIC_IMPORT_CANDIDATE_MEMBERS} should be
+   * mutually exclusive with identifiers listed here.
+   *
+   * <p>This should be a superset of the identifiers flagged by {@link
+   * com.google.errorprone.bugpatterns.BadImport}.
+   */
+  static final ImmutableSet<String> BAD_STATIC_IMPORT_CANDIDATE_IDENTIFIERS =
+      ImmutableSet.of(
+          "builder",
+          "create",
+          "copyOf",
+          "from",
+          "getDefaultInstance",
+          "INSTANCE",
+          "MIN",
+          "MIN_VALUE",
+          "MAX",
+          "newBuilder",
+          "of",
+          "valueOf");
+
+  /** Instantiates a new {@link BadStaticImport} instance. */
+  public BadStaticImport() {}
+
+  @Override
+  public Description matchIdentifier(IdentifierTree tree, VisitorState state) {
+    if (isMatch(tree, state)) {
+      return getDescription(tree, state);
+    }
+
+    return Description.NO_MATCH;
+  }
+
+  private Description getDescription(IdentifierTree tree, VisitorState state) {
+    Symbol symbol = ASTHelpers.getSymbol(tree);
+    SuggestedFix.Builder builder =
+        SuggestedFix.builder().removeStaticImport(getImportToRemove(symbol));
```

Comment:
Hmm, I'm curious if we can always do this 🤔. Removing an import _can_ be dangerous. This is usually a task for the `RemoveUnusedImport` check in a subsequent run. 

However, I'm thinking that we can actually get away with this one. What if there are two identifiers statically imported that we update, is there a possibility that one of the two _cannot_ be imported 🤔. If there is a possibility that can happen, we cannot remove the static import.

Reply:
Good question. I'm thinking the following:

Say we have an identifier `X` that with two occurrences (or more) in the same file. IIUC, the situation we want to avoid is the `isMatch` method returning true in one place and false in another.

However, for that to happen, one of the following must return different results for the occurrences:
- `symbol.owner != TYP` Can't have different owners
- `isDefinedInThisFile(...)` They can't be defined in different files
- `isIdentifierStaticallyImported(...)` Since it is the same identifier we will have the same result
- `isExempted` and `isCandidate` Will return the same results

I guess it boils down to the fact that we never use the surrounding context of the identifier to make a decision. WDYT?